probe-rs/src/architecture/arm/ap/mod.rs

```diff
@@ -13,32 +13,44 @@ pub use memory_ap::{
 use super::Register;
 use thiserror::Error;
 
-#[derive(Debug, Error, PartialEq)]
+#[derive(Debug, Error)]
 pub enum AccessPortError {
-    #[error("Invalid Access PortType Number")]
-    InvalidAccessPortNumber,
-    #[error("Misaligned memory access")]
-    MemoryNotAligned,
+    #[error("Failed to access address 0x{0:08x} as it is not aligned to 4 bytes.")]
```

Comment:
Are we sure that the address always has to be aligned to 4 bytes? I think we could have cases where we need e.g. 2 byte alignment.

Reply:
This error is only issued when we do 32bit reads where it is needed apparently. For 8bit reads this works ofc. We do not implement 16bit reads yet. A user always has to read 32bits anyways atm, so some of the read always goes to waste :)

Reply:
Yes, right know we don't cases where we need other alignment. But I would prefer to keep the error generic.

And you can read 2 16 bit words, so there is no need for wasting anything.